bin/migrate-oats-data/applications/decisions/application_decisions_init.py

```diff
@@ -0,0 +1,202 @@
+                    oats_alr_appl_decision_id
+                )
+                VALUES (
+                    %(audit_created_by)s,
+                    %(created_at)s,
+                    %(date)s,
+                    %(decision_description)s,
+                    false,
+                    false,
+                    %(outcome_code)s,
+                    %(rescinded_comment)s,
+                    %(rescinded_date)s,
+                    %(resolution_number)s,
+                    %(resolution_year)s,
+                    true,
+                    %(application_uuid)s,
+                    %(is_subject_to_conditions)s,
+                    %(oats_alr_appl_decision_id)s
+                )
+                ON CONFLICT ("resolution_number", "resolution_year")
+                    WHERE "audit_deleted_date_at" is null and "resolution_number" is not NULL
+                DO UPDATE SET 
+                date = COALESCE(EXCLUDED.date, alcs.application_decision.date),
+                outcome_code = COALESCE(EXCLUDED.outcome_code, alcs.application_decision.outcome_code),
+                decision_description = COALESCE(EXCLUDED.decision_description, alcs.application_decision.decision_description),
+                rescinded_comment =  COALESCE(EXCLUDED.rescinded_comment, alcs.application_decision.rescinded_comment),
+                rescinded_date =  COALESCE(EXCLUDED.rescinded_date, alcs.application_decision.rescinded_date),
+                is_subject_to_conditions = COALESCE(EXCLUDED.is_subject_to_conditions, alcs.application_decision.is_subject_to_conditions),
+                oats_alr_appl_decision_id = EXCLUDED.oats_alr_appl_decision_id;
+    """
+    return query
+
+
+def _prepare_oats_alr_applications_data(row_data_list):
+    data_list = []
+    for row in row_data_list:
+        resolution_number, year = _map_resolution_number_and_year(row)
+        mapped_row = {
+            "audit_created_by": OATS_ETL_USER,
+            "created_at": add_timezone_and_keep_date_part(row.get("when_created")),
+            "date": add_timezone_and_keep_date_part(row.get("decision_date")),
+            "decision_description": row.get("decision_desc"),
+            "decision_maker": "CEO",
+            "is_draft": False,
+            "outcome_code": _map_outcome_code(row),
+            "rescinded_comment": row.get("rescinded_comments"),
+            "rescinded_date": add_timezone_and_keep_date_part(
+                row.get("rescinded_date")
+            ),
+            "resolution_number": resolution_number,
+            "resolution_year": year,
+            "was_released": True,
+            "application_uuid": row.get("app_uuid"),
+            "is_subject_to_conditions": _map_is_subject_to_conditions(row),
+            "oats_alr_appl_decision_id": row.get("alr_appl_decision_id"),
+        }
+        data_list.append(mapped_row)
+
+    return data_list
+
+
+def _map_outcome_code(row):
+    if row.get("rescinded_date", None):
+        return "RESC"
```

Comment:
there is a AlcsDecisionOutcomeCodeEnum in common that could be used for this mapping